test/postgresql.jl

```diff
@@ -0,0 +1,78 @@
+        ODBC.execute!(dsn, "create table test1
+                            (test_bigint bigint,
+                             test_decimal decimal,
+                             test_int integer,
+                             test_numeric numeric,
+                             test_smallint smallint,
+                             test_float real,
+                             test_real double precision,
+                             test_money money,
+                             test_date date,
+                             test_timestamp timestamp,
+                             test_time time,
+                             test_char char(1),
+                             test_varchar varchar(16),
+                             test_bytea bytea,
+                             test_boolean boolean,
+                             test_text text,
+                             test_array integer[]
+                            )")
+        data = ODBC.query(dsn, "select * from information_schema.tables where table_name = 'test1'")
+        showall(data)
+        ODBC.execute!(dsn, "insert into test1 VALUES
+                            (1, -- bigint,
+                             1.2, -- decimal,
+                             2, -- integer,
+                             1.4, -- numeric,
+                             3, -- smallint,
+                             1.6, -- real,
+                             1.8, -- double precision,
+                             2.0, -- money,
+                             '2016-01-01', -- date,
+                             '2016-01-01 01:01:01', -- timestamp,
+                             '01:01:01', -- time,
+                             'A', -- char(1),
+                             'hey there sailor', -- varchar(16),
+                             NULL, -- bytea,
+                             TRUE, -- boolean,
+                             'hey there abraham', -- text
+                             ARRAY[1, 2, 3] -- integer array
+                            )")
+        source = ODBC.Source(dsn, "select * from test1")
+        data = ODBC.query(source)
+        @test size(Data.schema(data)) == (1,17)
+        @test Data.types(Data.schema(data)) == (
+          Union{Int64, Missing}
+         ,Union{DecFP.Dec64, Missing}
+         ,Union{Int32, Missing}
+         ,Union{DecFP.Dec64, Missing}
+         ,Union{Int16, Missing}
+         ,Union{Float32, Missing}
+         ,Union{Float64, Missing}
+         ,Union{Float64, Missing}
+         ,Union{ODBC.API.SQLDate, Missing}
+         ,Union{ODBC.API.SQLTimestamp, Missing}
+         ,Union{ODBC.API.SQLTime, Missing}
+         ,Union{WeakRefStrings.WeakRefString{UInt8}, Missing}
+         ,Union{WeakRefStrings.WeakRefString{UInt8}, Missing}
+         ,Union{Array{UInt8,1}, Missing}
+         ,Union{WeakRefStrings.WeakRefString{UInt8}, Missing}
+         ,Union{String, Missing}
+         ,Union{WeakRefStrings.WeakRefString{UInt8}, Missing})
```

Comment:
Instead of setting an upper-bound on WeakRefStrings. You could alternatively change this to:
```julia
types = Data.types(Data.schema(data))
@test Tuple{types...} <: Tuple{
    Union{Int64, Missing},
    Union{DecFP.Dec64, Missing},
    Union{Int32, Missing},
    Union{DecFP.Dec64, Missing},
    Union{Int16, Missing},
    Union{Float32, Missing},
    Union{Float64, Missing},
    Union{Float64, Missing},
    Union{ODBC.API.SQLDate, Missing},
    Union{ODBC.API.SQLTimestamp, Missing},
    Union{ODBC.API.SQLTime, Missing},
    Union{AbstractString, Missing},
    Union{AbstractString, Missing},
    Union{Array{UInt8,1}, Missings.Missing},
    Union{AbstractString, Missing},
    Union{String, Missing},
    Union{AbstractString, Missing},
}
```

Reply:
We'll definitely want to keep this as a separate commit

Reply:
@omus So it seems like the updated WeakRefStrings has more things that break ODBC:

```
prepared statement: Error During Test
  Got an exception of type ErrorException outside of a @test
  you must implement endof(WeakRefString{UInt8})
  Stacktrace:
   [1] length at ./strings/basic.jl:94 [inlined]
   [2] sqllength(::WeakRefString{UInt8}) at /Users/username/.julia/v0.6/ODBC/src/Source.jl:45
   [3] execute!(::ODBC.Statement, ::Array{Any,1}) at /Users/username/.julia/v0.6/ODBC/src/Source.jl:71
   [4] macro expansion at /Users/username/.julia/v0.6/ODBC/test/mysql.jl:209 [inlined]
   [5] macro expansion at ./test.jl:860 [inlined]
   [6] macro expansion at /Users/username/.julia/v0.6/ODBC/test/mysql.jl:204 [inlined]
   [7] macro expansion at ./test.jl:860 [inlined]
   [8] anonymous at ./<missing>:?
   [9] include_from_node1(::String) at ./loading.jl:576
   [10] include(::String) at ./sysimg.jl:14
   [11] macro expansion at /Users/username/.julia/v0.6/ODBC/test/runtests.jl:53 [inlined]
   [12] macro expansion at ./test.jl:860 [inlined]
   [13] anonymous at ./<missing>:?
   [14] include_from_node1(::String) at ./loading.jl:576
   [15] include(::String) at ./sysimg.jl:14
   [16] process_options(::Base.JLOptions) at ./client.jl:305
   [17] _start() at ./client.jl:371
```

Reply:
Hmmm, ok. I guess this is another issue to file

Reply:
Will do